pkg/controller/core.oam.dev/v1alpha2/core/scopes/healthscope/healthscope_controller.go

```diff
@@ -257,3 +259,15 @@ func (r *Reconciler) GetScopeHealthStatus(ctx context.Context, healthScope *v1al
 
 	return scopeCondition, workloadHealthConditions
 }
+
+// UpdateStatus updates v1alpha2.HealthScope's Status with retry.RetryOnConflict
+func (r *Reconciler) UpdateStatus(ctx context.Context, hs *v1alpha2.HealthScope, opts ...client.UpdateOption) error {
```

Comment:
is there a way to not repeat this over and over?

Reply:
That's because Go don't have the generic.

Reply:
we could use a common function